network/room (server): add message/join/leave/kick/ban/unban/game name logging

vvanelslande opened the pull request:
This allows you to see what happens in your rooms without joining them if you use citra-room.

Requested by `Drgn#6630` in the MHXX Discord server.

Sample logs:
```
[  10.294862] Network <Info> network\room.cpp:Network::Room::RoomImpl::SendStatusMessage:787: Test (vvanelslande) has joined.
[  12.900549] Network <Info> network\room.cpp:Network::Room::RoomImpl::HandleChatPacket:955: Test (vvanelslande): Test                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
[  17.311502] Network <Info> network\room.cpp:Network::Room::RoomImpl::SendStatusMessage:794: Test (vvanelslande) has left.
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/citra-emu/citra/4987)
<!-- Reviewable:end -->

chris062689 commented:
> is using the same file for SDL, Qt and dedicated room ok?

I would imagine this logging would only be useful in dedicated-room, but let others chime in as well.

vvanelslande commented:
> > is using the same file for SDL, Qt and dedicated room ok?
> 
> I would imagine this logging would only be useful in dedicated-room, but let others chime in as well.

I meant "is using the same log file for SDL, Qt and dedicated room ok?"

vvanelslande commented:
Added log filename option (default: user/log/citra-room.log), requested by `ANTHENA#0474` in the MHXX Discord server.

Added game name logging, requested by `ANTHENA#0474` in the MHXX Discord server.

zhaowenlan1779 commented on src/dedicated_room/citra-room.cpp:
```diff
@@ -271,6 +289,13 @@ int main(int argc, char** argv) {
+    Common::DetachedTasks detached_tasks;
+
+    // This is just to be able to link against core
+    gladLoadGL();
```
Why are these two lines moved?

zhaowenlan1779 commented on src/dedicated_room/citra-room.cpp:
```diff
@@ -162,14 +176,15 @@ int main(int argc, char** argv) {
 
     while (optind < argc) {
-        int arg = getopt_long(argc, argv, "n:d:p:m:w:g:u:t:a:i:hv", long_options, &option_index);
+        int arg = getopt_long(argc, argv, "n:d:p:m:w:g:u:t:a:i:f:hv", long_options, &option_index);
```
I'd prefer something like `-l` for `log-file`. `-f` isn't as clear IMO.

zhaowenlan1779 commented on src/network/room.cpp:
```diff
@@ -778,6 +778,42 @@ void Room::RoomImpl::SendStatusMessage(StatusMessageTypes type, const std::strin
+            LOG_INFO(Network, "{} has been banned.", nickname);
+        } else {
+            LOG_INFO(Network, "{} ({}) has been banned.", nickname, username);
+        }
```
If you are going to use so many `if`s, I think you may as well just do this and use it in every `case`.
```cpp
const std::string display_name = username.empty() ? nickname : fmt::format("{} ({})", nickname, username);
...
```

zhaowenlan1779 commented on src/network/room.cpp:
```diff
@@ -930,6 +973,21 @@ void Room::RoomImpl::HandleGameNamePacket(const ENetEvent* event) {
+                    LOG_INFO(Network, "{} ({}) is playing {}", member->nickname,
+                             member->user_data.username, game_info.name);
+                }
+            }
```
ditto. I would prefer having a variable instead of many `if`s
